src/components/NotificationModal.tsx

```diff
@@ -0,0 +1,229 @@
+                <Input onChange={(e) => handleChange(e, setState)} name="carNickname" id="car-nickname" value={state.userInput.carNickname} placeholder="Enter car nickname" />
+                {errors.find(error => error.path.includes('carNickname')) && <p className="text-xs text-red-500">{getErrorMessage('carNickname')}</p>}
+              </div>
+              <div className="grid gap-2">
+                {!state.notifUnsocHours.acceptedUnsocialHours &&
+                  <>
+                    <Label htmlFor="notification-time">Notification Time</Label>
+                    <Select name="notification-time" defaultValue="1440" onValueChange={
+                      (e) => {
+                        const updatedNotifUnsocHours = calculateUnsociableHoursSuggestionDaybeforeOrSameday(
+                          state.userInput.notificationDate,
+                          state.notifUnsocHours
+                        )
+                        setState(prev => ({
+                          ...prev,
+                          notifUnsocHours: updatedNotifUnsocHours
+                        }))
+                        handleSelectionChange(e, setState, location)
+                      }
+                    }>
+                      <SelectTrigger className="w-full">
+                        <SelectValue placeholder="Select time" />
+                      </SelectTrigger>
+                      <SelectContent>
+                        {getFilteredOptions().map(option => (
+                          <SelectItem key={option.value} value={option.value}>
+                            {option.label}
+                          </SelectItem>
+                        ))}
+                      </SelectContent>
+                    </Select>
+                  </>
+                }
+                {state.notifUnsocHours.suggestUnsocialHours && (
+                  <div className={`flex space-x-2 items-center ${!state.notifUnsocHours.acceptedUnsocialHours ? 'bg-red-100 p-2 rounded-sm' : ''}`}>
+                    {!state.notifUnsocHours.acceptedUnsocialHours &&
+                      <>
+                        <div className="text-xs m-2 text-black">
+                          Your requested notification time lands during unsociable hours, on <span className="font-bold">{
+                            state.userInput.notificationDate?.toLocaleString(timestampFormatOpts)
+                          }</span>. Shall we notify you at <span className="font-bold">20:00</span> instead on the {state.notifUnsocHours.dayBefore ? "day before" : "same day"}?
+                        </div>
+
+                        <Button
+                          id="unsocialHours"
+                          onClick={handleAcceptUnsocHours}
+                        >
+                          Sure!
+                        </Button>
+                      </>
+                    }
+                  </div>
```

Comment:
A short term solution for the problem of unsocial hour adjustments pushing the notification into the past. Unsocial hour adjustments will only be suggested if the cleaning isn't within the next 24h.